### PR TITLE
Treat "." and ".." as relative module names

### LINKED ISSUE #9651: Node Module Resolution does not treat ".." or "." as a relative module name
**TypeScript Version:**  2.0.0-dev.20160707

**Code**

```ts
import * as x from ".."; // incorrect resolution
import * as y from "../"; // correct resolution
```

**Expected behavior:**

Node module resolution treats ".." or "." as a relative module name, and finds the "index.ts" in the parent folder.

**Actual behavior:** 

Node module resolution treats ".." or "." as a non-relative module name and attempts to resolve it from "node_modules".

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #9651

### CODE REVIEW

File: src/compiler/utilities.ts [hunk 1]
```diff
@@ -1218,7 +1218,23 @@ namespace ts {
     export function isExternalModuleNameRelative(moduleName: string): boolean {
         // TypeScript 1.0 spec (April 2014): 11.2.1
         // An external module name is "relative" if the first term is "." or "..".
-        return moduleName.substr(0, 2) === "./" || moduleName.substr(0, 3) === "../" || moduleName.substr(0, 2) === ".\\" || moduleName.substr(0, 3) === "..\\";
```
Comment (sandersn) [inline src/compiler/utilities.ts]: why not use ` /^\.\.?($|[\\/])/.test(moduleName)`? It's a regex, but I think it's still easier to read than 16 lines of imperative code.

**Comment by sandersn** [comment]
:+1:

**Comment by andy-ms** [comment]
One more 8e679b7

File: src/compiler/core.ts [hunk 2]
```diff
@@ -915,6 +913,17 @@ namespace ts {
         return false;
     }
 
+    // Should act like String.prototype.startsWith
+    export function stringStartsWith(s: string, start: string): boolean {
```
Comment (vladima) [inline src/compiler/core.ts]: we already have `startsWith` and `endsWith` in `utilities.ts`

File: tests/cases/compiler/relativeModuleWithoutSlash.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// @Filename: index.ts
```
Comment (vladima) [inline tests/cases/compiler/relativeModuleWithoutSlash.ts]: add `@traceResolution: true` to capture resolution trace in baseline

**Pull request merged**